tests/server/api/v0/test_users.py

```diff
@@ -37,6 +43,43 @@ def test_me_without_authentication(client: TestClient):
     assert response.status_code == 401
 
 
+def test_admin_user_will_list_all_workspaces(client: TestClient, db: Session):
+    admin = AdminFactory.create()
+    annotator = AnnotatorFactory.create()
+    ws1, ws2 = WorkspaceFactory.create(), WorkspaceFactory.create()
+
+    response = client.get("/api/me", headers={API_KEY_HEADER_NAME: admin.api_key})
+    response_body = response.json()
+
+    assert response_body["id"] == str(admin.id)
+    assert ws1.name in response_body["workspaces"]
+    assert ws2.name in response_body["workspaces"]
+
+    response = client.get("/api/me", headers={API_KEY_HEADER_NAME: annotator.api_key})
+    response_body = response.json()
+
+    assert response_body["id"] == str(annotator.id)
+    assert not response_body["workspaces"]
+
+
+def test_admin_user_with_assigned_workspace_will_list_all_workspaces_without_duplicates(
+    client: TestClient, db: Session
+):
+    admin = AdminFactory.create()
+    ws1, ws2 = WorkspaceFactory.create(name="ws01"), WorkspaceFactory.create(name="ws02")
+    WorkspaceUserFactory.create(user_id=admin.id, workspace_id=ws1.id)
+
+    response = client.get("/api/me", headers={API_KEY_HEADER_NAME: admin.api_key})
+    response_body = response.json()
+
+    assert response_body["id"] == str(admin.id)
+
+    user_workspace_names = response_body["workspaces"]
+    user_workspace_names.sort()
+
+    assert user_workspace_names == [ws1.name, ws2.name]
+
+
```

Comment:
I think you should create one for admin, another for non-admin (annotator) and that's it:

```suggestion
def test_me_as_admin(client: TestClient, db: Session):
    admin = AdminFactory.create(
        workspaces=[
            WorkspaceFactory.build(name="workspace-a"),
            WorkspaceFactory.build(name="workspace-b")
        ]
    )
    WorkspaceFactory.create(name="workspace-c")

    response = client.get("/api/me", headers={API_KEY_HEADER_NAME: admin.api_key})

    assert response.status_code == 200

    response_body = response.json()
    assert response_body["id"] == str(admin.id)
    assert response_body["workspaces"] == ["workspace-a", "workspace-b", "workspace-c"]


def test_me_as_annotator(client: TestClient, db: Session):
    annotator = AnnotatorFactory.create(
        workspaces=[
            WorkspaceFactory.build(name="workspace-a"),
            WorkspaceFactory.build(name="workspace-b")
        ]
    )
    WorkspaceFactory.create(name="workspace-c")

    response = client.get("/api/me", headers={API_KEY_HEADER_NAME: annotator.api_key})

    assert response.status_code == 200

    response_body = response.json()
    assert response_body["id"] == str(annotator.id)
    assert response_body["workspaces"] == ["workspace-a", "workspace-b"]


```